lib/api_auth/base.rb

```diff
@@ -78,23 +80,31 @@ def request_too_old?(headers)
       end
     end
 
-    def signatures_match?(headers, secret_key)
+    def signatures_match?(headers, secret_key, options)
+      options = options.merge(:with_http_method => true)
       if match_data = parse_auth_header(headers.authorization_header)
         hmac = match_data[2]
-        hmac == hmac_signature(headers, secret_key)
+        return (
```

Comment:
These parens looked weird so I tried removing them. Your tests pass without them, so either these aren't needed or this piece of functionality is under-tested.